Update to use "twilight" name instead of "dawn" and use latest versions of dependencies

liminalitythree opened the pull request:
This PR
* changes the name of the crate from `dawn-http-proxy` to `twilight-http-proxy`
* updates several alpha dependencies to the stable versions
* removes the `futures-preview` dependency, which was used for something that the latest version of `hyper` can do on it's own
* updates from `dawn_http` to `twilight_http`
* updates the README.md, changing `dawn` to `twilight` and fixing the example to work with the latest version of `twilight_http`

liminalitythree committed:
Update to use latest versions of dependencies and change name from dawn-http-proxy to twilight-http-proxy

liminalitythree committed:
Update README.md to use new "twilight" name

liminalitythree committed:
Remove unused "stream" feature from tokio

Erk- commented on Cargo.toml:
```diff
@@ -1,16 +1,15 @@
 [package]
 authors = ["Zeyla Hellyer <zeyla@hellyer.dev>"]
```
Update this to "Twilight Contributors" as the main libraries

Erk- commented on Cargo.toml:
```diff
@@ -1,16 +1,15 @@
+hyper = "0.13"
+http = "0.2"
 log = "0.4"
 pretty_env_logger = "0.3"
```
Switch out with tracing_subscriber and maybe tracing_log as well

Erk- reviewed (changes requested):
A few nitpicks

Erk- commented:
Could you also update the license in a similar way to the one in the main repository?

liminalitythree committed:
Update license and authors

Update LICENSE.md and the authors portion of Cargo.toml to use "Twilight Contributors"

liminalitythree committed:
Replace log with tracing

spring4175 commented on Cargo.toml:
```diff
@@ -1,16 +1,16 @@
+reqwest = "0.10"
 snafu = "0.5"
-tokio = "0.2.0-alpha.4"
+tokio = { version = "0.2", features = ["macros"] }
```
Does this [need `rt-threaded`](https://docs.rs/tokio/0.2.22/tokio/attr.main.html)? It might be getting enabled by some dependency but it should be explicitly enabled anyway

Erk- replied:
It only *needs* rt-core, if rt-threaded is enabled it uses multiple threads.

spring4175 replied:
rt-core it is then

spring4175 reviewed (changes requested):
LGTM other than that one nit (if I'm not wrong). Was rustfmt run on this? The response builder being on two lines looks like something it wouldn't do

liminalitythree committed:
run rustfmt

spring4175 reviewed (commented):
Oh, kind of assumed it was already rustfmt'd, oh well. Or maybe it's just been a while since. Just needs the rt-core feature selection to tokio in the Cargo.toml and then it should be good to go :)

liminalitythree committed:
Add rt-core feature to tokio dependency

spring4175 reviewed (approved):
lgtm

spring4175 commented:
Thanks!
